internal/types/etcd_storage.go

```diff
@@ -23,6 +25,17 @@ func ParseETCDStorageType(s string) ETCDStorageType {
 		return storageType
 	}
 
-	// TODO: we have to decide what to do in this situation
-	return defaultETCDStorageType
+	panic(fmt.Errorf("unsupported storage type %s", s))
+}
+
+// ParseETCDEndpoint returns the default ETCD endpoints used to interact with the Tenant Control Plane backing storage.
+func ParseETCDEndpoint(conf *viper.Viper) string {
+	switch ParseETCDStorageType(conf.GetString("etcd-storage-type")) {
+	case ETCD:
+		return conf.GetString("etcd-endpoints")
```

Comment:
you could add the `https` protocol over here

Reply:
It would work: `etcd-endpoints` is a string flag separated by a comma.

It would require splitting by comma, checking if there's a protocol, and adding it in case of missing.

Since etcd backing storage requires HTTPS connection I think it's safe to avoid it, don't you think?